Fix shfmt and shellcheck warnings

coiby opened the pull request:
shfmt and shellcheck have found some issues in the kdump-utils code. Although current majority of the issues there are real problems, it's good to fix them so news issues can be caught more easily.

## Summary by Sourcery

Enforce consistent shell formatting and resolve ShellCheck warnings across kdump-utils by reformatting scripts, adding missing quotes, and adjusting CI and build dependencies.

Enhancements:
- Reformat all shell scripts using shfmt to standardize indentation, function definitions, and bracket spacing
- Quote variable expansions and command substitutions throughout code and tests to prevent word-splitting and ShellCheck errors
- Insert shellcheck disable pragmas and source annotations where necessary to suppress false positives
- Update test and spec files to properly quote parameters and accommodate shellspec directives

Build:
- Add gawk as an RPM build dependency in the spec file

CI:
- Install gawk in CI runner and refine shellcheck invocation to include spec files
- Disable specific ShellCheck checks in CI for unit test patterns

licliu commented on kdump-dep-generator.sh:
```diff
@@ -11,13 +11,13 @@
-    mkdir -p $kdump_wants
-    ln -sf $systemd_dir/network-online.target $kdump_wants/
+	mkdir -p "$kdump_wants"
+	ln -sf $systemd_dir/network-online.target "$kdump_wants"/
```
+1

licliu commented on kdump-lib-initramfs.sh:
```diff
@@ -4,10 +4,12 @@
+# shellcheck disable=SC2034
 DEFAULT_SSHKEY="/root/.ssh/kdump_id_rsa"
 KDUMP_CONFIG_FILE="/etc/kdump.conf"
-FENCE_KDUMP_CONFIG_FILE="/etc/sysconfig/fence_kdump"
```
The default value of FENCE_KDUMP_CONFIG_FILE was removed instead of being moved to kdump-lib.sh.

coiby replied:
Good catch! Fixed in new version.

licliu commented on kdump-lib.sh:
```diff
@@ -764,7 +768,7 @@ prepare_cmdline()
 
 	id=$(get_bootcpu_apicid)
-	if [[ -n "${id}" ]]; then
+	if [[ -n ${id} ]]; then
```
Is there any difference between using `$id` and `${id}` here?

coiby commented:
Hi @licliu,

Thanks for carefully reviewing the PR! I've pushed a new version which also 1) rebases onto latest main branch and 2) fixed [an issue caught by Gemini early](https://github.com/coiby/kdump-utils/pull/14#discussion_r2122402079) but I forgot to fix besides addressing your feedback ([diff change](https://github.com/rhkdump/kdump-utils/compare/7fa77f8705a80e2112417e397dcfd4181d1165f5..04dd80332e61001f32b1fe43f3ec9bd16d1daee2)) .

licliu reviewed (approved):
Thanks @coiby, this version looks good to me.
